src/server/app/main.py

```diff
@@ -0,0 +1,150 @@
+from datetime import datetime
+from typing import Annotated, List, Literal, Optional
+
+from fastapi import FastAPI, HTTPException, Query
+from pydantic import PositiveInt
+from stac_fastapi.types.rfc3339 import str_to_interval
+from stac_pydantic.shared import BBox
+
+from app.catalog_collection_search import CatalogCollectionSearch, DatetimeInterval
+from app.catalog_search_service import CatalogSearchService
+from app.cmr_collection_search import CMRCollectionSearch
+from app.config import Settings
+from app.models import SearchResponse
+from app.stac_api_collection_search import STACAPICollectionSearch
+
+invalid_bbox = HTTPException(status_code=400, detail="Invalid bbox")
+
+
+def str_to_bbox(bbox_str: Optional[str]) -> Optional[BBox]:
+    """Convert string to BBox based on , delimiter."""
+    if not bbox_str:
+        return None
+
+    try:
+        t = tuple(float(v) for v in bbox_str.split(","))
+    except ValueError:
+        raise invalid_bbox
+    if not len(t) == 4:
+        raise invalid_bbox
+
+    return t
```

Comment:
This works better with type checking:

```suggestion
    try:
        x0, y0, x1, y1 = map(float, bbox_str.split(","))
    except ValueError:
        raise invalid_bbox from None

    return x0, y0, x1, y1
```